sdk/core/azure-core/src/main/java/com/azure/core/util/paging/SyncPageRetriever.java

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.util.paging;
+
+import com.azure.core.util.IterableStream;
+
+/**
+ * This class handles retrieving pages.
+ *
+ * @param <C> Type of the continuation token.
+ * @param <P> the page elements type
+ */
+@FunctionalInterface
+public interface SyncPageRetriever<C, P> {
+
+    /**
+     * Retrieves one or more pages starting from the page identified by the given continuation token.
+     *
+     * @param continuationToken Token identifying which page to retrieve, passing {@code null} indicates to retrieve
+     * the first page.
+     * @param pageSize The number of items to retrieve per page, passing {@code null} will use the source's default
+     * page size.
+     * @return A {@link IterableStream} that emits one or more pages.
+     */
+    IterableStream<P> getIterable(C continuationToken, Integer pageSize);
```

Comment:
Do we need this to return `IterableStream`? For synchronous APIs, do we request more than one page per call to `hasNext()`?

Reply:
I agree with this, the asynchronous one returning `Flux` was out of necessity/a mistake, ideally the contract for paging is that only one page should be requested at any time. I'd also want to give this API a better name of `getPage` as this is what it's actually doing.